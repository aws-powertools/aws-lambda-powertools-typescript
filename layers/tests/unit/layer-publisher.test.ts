/**
 * Test LayerPublisherStack class
 *
 * @group unit/layers/all
 */

import { App } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import {
  LayerPublisherStack
} from '../../src/layer-publisher-stack';

describe('Class: LayerPublisherStack', () => {

  it('creates the stack with a layer in it', () => {

    // Prepare
    const app = new App();
    const stack = new LayerPublisherStack(app, 'MyTestStack', {
      layerName: 'AWSLambdaPowertoolsTypeScript',
      powertoolsPackageVersion: '1.0.1',
      ssmParameterLayerArn: '/layers/powertools-layer-arn',
    });

    // Act
    const template = Template.fromStack(stack);

    // Assess
    expect(template).toMatchSnapshot();

  });

});