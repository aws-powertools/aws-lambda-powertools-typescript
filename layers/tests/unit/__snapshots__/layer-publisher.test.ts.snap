// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Class: LayerPublisherStack creates the stack with a layer in it 1`] = `
{
  "Outputs": {
    "LatestLayerArn": {
      "Export": {
        "Name": "AWSLambdaPowertoolsTypeScript",
      },
      "Value": {
        "Ref": "LambdaPowertoolsLayer733361E9",
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "LambdaPowertoolsLayer733361E9": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "CompatibleRuntimes": [
          "nodejs14.x",
          "nodejs16.x",
          "nodejs18.x",
        ],
        "Content": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "02695fa751b4d79fb6eaea375c8f92416802c843204d917f10f9b43c8247672d.zip",
        },
        "Description": "AWS Lambda Powertools for TypeScript version 1.0.1",
        "LayerName": "AWSLambdaPowertoolsTypeScript",
      },
      "Type": "AWS::Lambda::LayerVersion",
      "UpdateReplacePolicy": "Retain",
    },
    "PublicLayerAccess": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "Action": "lambda:GetLayerVersion",
        "LayerVersionArn": {
          "Ref": "LambdaPowertoolsLayer733361E9",
        },
        "Principal": "*",
      },
      "Type": "AWS::Lambda::LayerVersionPermission",
      "UpdateReplacePolicy": "Retain",
    },
    "VersionArn99E177D3": {
      "Properties": {
        "Name": "/layers/powertools-layer-arn",
        "Type": "String",
        "Value": {
          "Ref": "LambdaPowertoolsLayer733361E9",
        },
      },
      "Type": "AWS::SSM::Parameter",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
